### PR TITLE
[CPM-129] Renovate courses page

### PR DESCRIPTION
**Pull request by Dinith1**
## Changes
- Before:
image

- After:
image

- I feel like the UI could be improved further
  - The course regulations table tags ('bubbles') could be better formatted (I left a TODO comment)
  - Add loading indicators

### AC:
- [x] The edit and cancel/save components should be consistent with the Programmes page
- [x] Dropdown Menu should hold editing and deleting
- [x] When searching courses and no results are found, the “No results found” text should be centered
- [x] Table should be renovated
  - [x] too large and boxy
  - [x] text is too large
  - [x] need to show semester in table
- [x] Course Info are only editable upon selecting edit within the dropdown menu (…) 
- [x] When editing, all fields become editable (like in Programmes page)
- [x] The Cancel & Save buttons should only appear when editing
- [x] Title of the course page (RHS) should be the course code
- [x] Add Course Button should be placed elsewhere (i.e in the top row of the table or next to the table/title) because it isn’t intuitive atm
- [ ] Split prereqs into quadrants?   [DON’T TOUCH UNTIL WENDY/RICHARD ARE DONE]
- [x] Background of LHS should be darker (similar to Programmes)

## Checklist

- [ ] Have tests been written?
- [x] Is the UI consistent?

## Requests
- [ ] Can somebody check the styling/CSS?

Addresses [CPM-129] 

Connects #y

[CPM-129]: https://momentum-agile.atlassian.net/browse/CPM-129

### CODE REVIEW

**Comment by FocalChord** [comment]
Damn we got dark mode ~_^

File: client/src/pages/Courses/editables/RegulationTable.js [hunk 1]
```diff
@@ -75,30 +83,51 @@ const RegulationTable = ({ name, updateCourse, course, regulationType }) => {
...
+                                    <OptionsMenu onEdit={() => handleRegTableEditClick(regulationList)} onDelete={() => handleRegTableDeleteClick(regulationList)} />
+                                </Flex>
+                            </Flex>
+
+                            <Flex direction="row" pb="5px">
+                                {regulationList.map((reg) => (
+                                    splitRegulation(reg).map((item, idx) => (
+                                        item.toLowerCase() === "and" || item.toLowerCase() === "or"
+                                            ? (
+                                                <Tag size="sm" key={idx} rounded="full" variant="solid" variantColor={item.toLowerCase() === "and" ? "gray" : "gray"} mr={1}>
```
Comment (FocalChord) [inline client/src/pages/Courses/editables/RegulationTable.js:105]: `item.toLowerCase() === "and" ? "gray" : "gray"` => `"gray"`

Comment (Dinith1) [reply client/src/pages/Courses/editables/RegulationTable.js:105]: Oof forgot to change that

File: client/src/pages/Courses/editables/TextField.js [hunk 2]
```diff
@@ -1,24 +1,99 @@
...
+                            <Flex>
+                                <Button
+                                    isDisabled={!value || !isValidCourseCode}
+                                    onClick={handlePrefill}
+                                    ml="10px"
+                                    bg={c.midnightBlue}
+                                    color={c.whiteGrey}
+                                    _hover={{ bg: !value || !isValidCourseCode ? null : c.lightBlue }}
+                                >
+                                    Auto-Generate Course &nbsp;
```
Comment (FocalChord) [inline client/src/pages/Courses/editables/TextField.js:87]: `&nbsp;` 🤔

Comment (Dinith1) [reply client/src/pages/Courses/editables/TextField.js:87]: Forgot to remove! 💯

File: client/src/pages/Courses/editables/RegulationTable.js (hunk 1 above)
Comment (Dinith1) [reply client/src/pages/Courses/editables/RegulationTable.js]: updated!

File: client/src/pages/Courses/editables/TextField.js (hunk 2 above)
Comment (Dinith1) [reply client/src/pages/Courses/editables/TextField.js]: updated!

File: client/src/pages/Courses/Courses.js [hunk 3]
```diff
@@ -23,94 +23,103 @@ const Courses = () => {
...
+                    />
                 </Flex>
 
+                <SearchBar searchCategory="Courses" value={searchTerm} onChange={(e) => setSearchTerm(e.target.value)} />
+
                 <Flex p={4}>
                     <Table
                         columns={columns}
-                        data={data}
+                        data={data.map(d => ({ ...d, semester: d.semester.join(", ") }))}
```
Comment (daltonlim) [inline client/src/pages/Courses/Courses.js:97]: u can leave the data untouched and change the semester accessor to accessor: d => d.semester.join(", "), in columns

File: client/src/pages/Courses/editables/RegulationTable.js [hunk 4]
```diff
@@ -75,30 +83,51 @@ const RegulationTable = ({ name, updateCourse, course, regulationType }) => {
...
+                                ))}
+                            </Flex>
+                        </Flex>
+
+                    ) : (
+                        <Button
+                            variantColor="blue"
+                            backgroundColor={c.midnightBlue}
+                            onClick={handleOpen}
+                            isDisabled={regulationList && regulationList.length > 0 ? true : false}
```
Comment (daltonlim) [inline client/src/pages/Courses/editables/RegulationTable.js:123]: Would you just be able to drop the ? true :false

**Review by daltonlim — APPROVED** [review]
code lgtm

File: client/src/pages/Courses/editables/RegulationTable.js (hunk 4 above)
Comment (Dinith1) [reply client/src/pages/Courses/editables/RegulationTable.js]: Removed!

File: client/src/pages/Courses/Courses.js (hunk 3 above)
Comment (Dinith1) [reply client/src/pages/Courses/Courses.js:97]: But the Students page uses the same table

Comment (daltonlim) [reply client/src/pages/Courses/Courses.js:97]: arent the defined columns different

Comment (daltonlim) [reply client/src/pages/Courses/Courses.js:97]: image
image
this seems to work

Comment (FocalChord) [reply client/src/pages/Courses/Courses.js:97]: Agree that we shouldn't be directly manipulating the data ^

Comment (Dinith1) [reply client/src/pages/Courses/Courses.js:97]: Ok i'll try it

**Comment by daltonlim** [comment]
@pull-dog up

**Review by FocalChord — APPROVED** [review]
Absolute Ripper

**Pull request merged**